modules/nf-core/gtdbtk/classifywf/main.nf

```diff
@@ -1,12 +1,14 @@
+def VERSION = '2.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

Comment:
This should go back to line 33, which is the new standard

Reply:
I was also wondering, but then the variable is not recognized in the stub scope :(

```bash
$ nf-core modules test gtdbtk/classifywf

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.6 - https://nf-co.re


INFO     Press enter to use default values (shown in brackets) or type your own responses                         module_test.py:69
? Choose software profile Singularity
INFO     Setting environment variable '$PROFILE' to 'singularity'                                                module_test.py:153
──────────────────────────────────────────────────────── gtdbtk/classifywf ────────────────────────────────────────────────────────
INFO     Running pytest for module 'gtdbtk/classifywf'                                                           module_test.py:180
======================================================= test session starts =======================================================
platform linux -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /r1/people/maxime_borry/tmp/nf-core-modules, configfile: pytest.ini
plugins: workflow-1.6.0
collecting ...
collected 1155 items

gtdbtk classifywf:
	command:   nextflow run ./tests/modules/nf-core/gtdbtk/classifywf -entry test_gtdbtk_classifywf -c ./tests/config/nextflow.config -stub-run -c ./tests/modules/nf-core/gtdbtk/classifywf/nextflow.config
	directory: /tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf
	stdout:    /tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/log.out
	stderr:    /tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/log.err
'gtdbtk classifywf' done.

tests/test_versions_yml.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [  8%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 18%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 29%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 40%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssFsssssssssssssssssss [ 50%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 61%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 72%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 82%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 93%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss                                                   [ 99%]
tests/modules/nf-core/gtdbtk/classifywf/test.yml FF                                                                         [100%] Keeping temporary directories and logs. Use '--kwd' or '--keep-workflow-wd' to disable this behaviour.


============================================================ FAILURES =============================================================
________________________________________ test_ensure_valid_version_yml[gtdbtk classifywf] _________________________________________

workflow_dir = PosixPath('/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf')

    @pytest.mark.workflow(*_get_workflow_names())
    def test_ensure_valid_version_yml(workflow_dir):
        workflow_dir = Path(workflow_dir)
        software_name = workflow_dir.name.split("_")[0].lower()
        try:
            versions_yml_file = workflow_dir / f"output/{software_name}/versions.yml"
>           versions_yml = versions_yml_file.read_text()

tests/test_versions_yml.py:31:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PosixPath('/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml'), encoding = None, errors = None

    def read_text(self, encoding=None, errors=None):
        """
        Open the file in text mode, read it, and close the file.
        """
>       with self.open(mode='r', encoding=encoding, errors=errors) as f:

/home/maxime_borry/.conda/envs/maxime/envs/nf-core/lib/python3.9/pathlib.py:1266:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PosixPath('/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml'), mode = 'r', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
>       return io.open(self, mode, buffering, encoding, errors, newline,
                       opener=self._opener)

/home/maxime_borry/.conda/envs/maxime/envs/nf-core/lib/python3.9/pathlib.py:1252:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PosixPath('/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml')
name = '/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)
E       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml'

/home/maxime_borry/.conda/envs/maxime/envs/nf-core/lib/python3.9/pathlib.py:1120: FileNotFoundError

During handling of the above exception, another exception occurred:

workflow_dir = PosixPath('/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf')

    @pytest.mark.workflow(*_get_workflow_names())
    def test_ensure_valid_version_yml(workflow_dir):
        workflow_dir = Path(workflow_dir)
        software_name = workflow_dir.name.split("_")[0].lower()
        try:
            versions_yml_file = workflow_dir / f"output/{software_name}/versions.yml"
            versions_yml = versions_yml_file.read_text()
        except FileNotFoundError:
>           raise AssertionError(
                dedent(
                    f"""\
                    `versions.yml` not found in the output directory.
                    Expected path: `{versions_yml_file}`

                    This can have multiple reasons:
                    * The test-workflow failed before a `versions.yml` could be generated.
                    * The workflow name in `test.yml` does not start with the tool name.
                    """
                )
            )
E           AssertionError: `versions.yml` not found in the output directory.
E           Expected path: `/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/versions.yml`
E
E           This can have multiple reasons:
E           * The test-workflow failed before a `versions.yml` could be generated.
E           * The workflow name in `test.yml` does not start with the tool name.

tests/test_versions_yml.py:33: AssertionError
__________________________________________________________ test session ___________________________________________________________
'/tmp/pytest_workflow_g82c8uj6/gtdbtk_classifywf/output/gtdbtk/gtdbtk.SPADES-test.stub.summary.tsv' does not exist while it should
__________________________________________________________ test session ___________________________________________________________
'gtdbtk classifywf' exited with exit code '1' instead of '0'.
======================================================== warnings summary =========================================================
../../../../../home/maxime_borry/.conda/envs/maxime/envs/nf-core/lib/python3.9/site-packages/pytest_workflow/plugin.py:122: 677 warnings
  /home/maxime_borry/.conda/envs/maxime/envs/nf-core/lib/python3.9/site-packages/pytest_workflow/plugin.py:122: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return YamlFile.from_parent(parent, fspath=path)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================== short test summary info =====================================================
FAILED tests/test_versions_yml.py::test_ensure_valid_version_yml[gtdbtk classifywf] - AssertionError: `versions.yml` not found i...
FAILED tests/modules/nf-core/gtdbtk/classifywf/test.yml::gtdbtk classifywf::output/gtdbtk/gtdbtk.SPADES-test.stub.summary.tsv::should exist
FAILED tests/modules/nf-core/gtdbtk/classifywf/test.yml::gtdbtk classifywf::exit code should be 0
========================================= 3 failed, 1152 skipped, 677 warnings in 17.16s ==========================================
```

Reply:
For future reference, the correct way is to move the env variables in the stub scope.